### PR TITLE
sql: remove redundant conditional paths

### LINKED ISSUE #111781: `if` with identical `then` and `else` blocks in `sql/sem/tree/backup.go`
https://github.com/cockroachdb/cockroach/blob/2da2993c4ad5a7b07a80ae42ba3f78e8018d9e64/pkg/sql/sem/tree/backup.go#L204-L211

https://github.com/cockroachdb/cockroach/blob/2da2993c4ad5a7b07a80ae42ba3f78e8018d9e64/pkg/sql/sem/tree/backup.go#L523-L531

In both conditionals, `then` and `else` blocks are identical.

(found by [revive](https://github.com/mgechev/revive) while testing it on cockroach code base)

### PR DESCRIPTION
**Pull request by Shikhar03Stark**
Fixes: #111781 
Remove redundant conditional path in sql package

### CODE REVIEW

**Comment by Shikhar03Stark** [comment]
bors r=stevendanna

**Comment by Shikhar03Stark** [comment]
Please CR
cc: @stevendanna

**Comment by Shikhar03Stark** [comment]
Hey, can you please verify or review my code @chavacava ?
TFTR!

**Comment by chavacava** [comment]
@Shikhar03Stark, sorry I'm not a contributor to this repository. I've just opened the issue.

**Review by stevendanna — COMMENTED** [review]
Thanks for the contribution! I gave this a quick look.

CI shows one of the tests is failing which I believe my comment should fix up for you. 

If you want to make sure by running this test locally you can do:

    ./dev test ./pkg/sql/parser --filter TestParseDatadriven/backup_restore

But this takes some initial setup: https://cockroachlabs.atlassian.net/wiki/spaces/CRDB/pages/181338446/Getting+and+building+CockroachDB+from+source so it is probably only worth it if you hope to make repeated contributions to the code base.

File: pkg/sql/sem/tree/backup.go [hunk 1]
```diff
@@ -200,16 +200,10 @@ func (node *Restore) Format(ctx *FmtCtx) {
...
-			ctx.WriteString(")")
-		} else {
-			ctx.WriteString(" WITH OPTIONS (")
-			ctx.FormatNode(&node.Options)
-			ctx.WriteString(")")
-		}
+	if !node.Options.IsDefault() && ctx.HasFlags(FmtHideConstants) {
+		ctx.WriteString(" WITH OPTIONS (")
+		ctx.FormatNode(&node.Options)
+		ctx.WriteString(")")
```
[the remark this answers is not in the document]
Comment (Shikhar03Stark) [reply pkg/sql/sem/tree/backup.go]: Got it. Made changes according to that 👍

**Comment by Shikhar03Stark** [comment]
Test passing on local, with above setup.
@stevendanna PTAL 😄

**Comment by Shikhar03Stark** [comment]
> @Shikhar03Stark Thanks for the contribution! The code looks good. Do you mind cleaning up the commit message a bit? After that, I think this will be good to go.
> 
> Something like:
> 
> ```
> sql: remove redundant conditional paths
> 
> Fixes #111781
> 
> Release note: None
> ```

Done, PTAL @stevendanna

**Comment by stevendanna** [comment]
🎉 LGTM

bors r=stevendanna

**Review by stevendanna — APPROVED** [review]

**Comment by stevendanna** [comment]
bors r=stevendanna

**Pull request merged**